openapi/api2pdf/types.bal

```diff
@@ -14,6 +14,112 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/http;
+
+# Client configuration details.
+@display {label: "Connection Config"}
+public type ConnectionConfig record {|
+    # The HTTP version understood by the client
+    http:HttpVersion httpVersion = http:HTTP_2_0;
+    # Configurations related to HTTP/1.x protocol
+    ClientHttp1Settings http1Settings?;
+    # Configurations related to HTTP/2 protocol
+    http:ClientHttp2Settings http2Settings?;
+    # The maximum time to wait (in seconds) for a response before closing the connection
+    decimal timeout = 60;
+    # The choice of setting `forwarded`/`x-forwarded` header
+    string forwarded = "disable";
+    # Configurations associated with request pooling
+    http:PoolConfiguration poolConfig?;
+    # HTTP caching related configurations
+    http:CacheConfig cache?;
+    # Specifies the way of handling compression (`accept-encoding`) header
+    http:Compression compression = http:COMPRESSION_AUTO;
+    # Configurations associated with the behaviour of the Circuit Breaker
+    http:CircuitBreakerConfig circuitBreaker?;
+    # Configurations associated with retrying
+    http:RetryConfig retryConfig?;
+    # Configurations associated with inbound response size limits
+    http:ResponseLimitConfigs responseLimits?;
+    # SSL/TLS-related options
+    http:ClientSecureSocket secureSocket?;
+    # Proxy server related options
+    http:ProxyConfig proxy?;
+    # Enables the inbound payload validation functionality which provided by the constraint package. Enabled by default
+    boolean validation = true;
+|};
+
+# Provides settings related to HTTP/1.x protocol.
+#
+# + keepAlive - Specifies whether to reuse a connection for multiple requests
+# + chunking - The chunking behaviour of the request
+# + proxy - Proxy server related options
+public type ClientHttp1Settings record {|
+    KeepAlive keepAlive = KEEPALIVE_AUTO;
+    Chunking chunking = CHUNKING_AUTO;
```

Comment:
```suggestion
    http:KeepAlive keepAlive = KEEPALIVE_AUTO;
    http:Chunking chunking = CHUNKING_AUTO;
```
Is this possible ?

Reply:
We are maintaining the records internally.

Reply:
But yes. This change we can do @aneeshafedo

Reply:
```public type ClientHttp1Settings record {|
    http:KeepAlive keepAlive = KEEPALIVE_AUTO;
    http:Chunking chunking = CHUNKING_AUTO;
    ProxyConfig proxy?;
|};```

Reply:
Aren't we adding metadata for these fields ?

Reply:
```suggestion
    http:KeepAlive keepAlive = http:KEEPALIVE_AUTO;
    http:Chunking chunking = http:CHUNKING_AUTO;
```
We can also go for this tight ? Then we do not need below types. 

```bal
# Defines the possible values for the keep-alive configuration in service and client endpoints.
public type KeepAlive KEEPALIVE_AUTO|KEEPALIVE_ALWAYS|KEEPALIVE_NEVER;

# Defines the possible values for the chunking configuration in HTTP services and clients.
#
# `AUTO`: If the payload is less than 8KB, content-length header is set in the outbound request/response,
# otherwise chunking header is set in the outbound request/response
# `ALWAYS`: Always set chunking header in the response
# `NEVER`: Never set the chunking header even if the payload is larger than 8KB in the outbound request/response
public type Chunking CHUNKING_AUTO|CHUNKING_ALWAYS|CHUNKING_NEVER;
```

Reply:
Yeah. Will do that way.

Reply:
Yes, let's reuse the `http` library record for this.

Reply:
+1 for this. I guess this does not impact to the configuability of the ConnectionConfig record.